File: mobile/lib/api/models/liked_track.dart
import "package:wavelength/api/models/playlist_track.dart";

class LikedTrack {
  final String likeId;
  final String email;
  final String title;
  final String thumbnail;
  final bool isExplicit;
  final String author;
  final String duration;
  final String videoId;
  final VideoType videoType;

  LikedTrack({
    required this.likeId,
    required this.email,
    required this.title,
    required this.thumbnail,
    required this.isExplicit,
    required this.author,
    required this.duration,
    required this.videoId,
    required this.videoType,
  });

  factory LikedTrack.fromJson(Map<String, dynamic> json) {
    return LikedTrack(
      likeId: json["likeId"] as String,
      email: json["email"] as String,
      title: json["title"] as String,
      thumbnail: json["thumbnail"] as String,
      isExplicit: json["isExplicit"] as bool,
      author: json["author"] as String,
      duration: json["duration"] as String,
      videoId: json["videoId"] as String,
      videoType: json["videoType"] == "track"
          ? VideoType.track
          : VideoType.uvideo,
    );
  }
}

File: server/services/gateway/models/liked_track.go
package models

import type_constants "wavelength/services/gateway/constants/types"

type LikedTrack struct {
	LikeId     string                           `json:"likeId"`
	Email      string                           `json:"email"`
	Title      string                           `json:"title"`
	Thumbnail  string                           `json:"thumbnail"`
	IsExplicit bool                             `json:"isExplicit"`
	Author     string                           `json:"author"`
	Duration   string                           `json:"duration"`
	VideoId    string                           `json:"videoId"`
	VideoType  type_constants.PlaylistTrackType `json:"videoType"`
}
